Snowflake.Data/Core/SFStatement.cs

```diff
@@ -295,15 +295,21 @@ internal void Cancel()
             if (request == null)
                 return;
 
-            var response = _restRequester.Post<NullDataResponse>(request);
-
-            if (response.success)
+            try
             {
-                logger.Info("Query cancellation succeed");
-            }
-            else
+                var response = _restRequester.Post<NullDataResponse>(request);
+                if (response.success)
+                {
+                    logger.Info("Query cancellation succeed");
+                }
+                else
+                {
+                    logger.Warn("Query cancellation failed.");
+                }
+            }            
+            catch (Exception ex)

```

Comment:
Swallowing errors on `Cancel()` could be considered a breaking change if anyone was relying on this behavior.